metagpt/ext/cr/actions/modify_code.py

```diff
@@ -1,8 +1,17 @@
+import datetime
+import itertools
+import os
+import re
+
 from unidiff import PatchSet
 
 from metagpt.actions.action import Action
-from metagpt.logs import logger
-from metagpt.utils.cr.cleaner import add_line_num_on_patch, rm_patch_useless_part
+from metagpt.utils.cr.cleaner import add_line_num_on_patch, rm_patch_useless_part, get_code_block_from_patch
```

Comment:
There is a DI tool version(metagpt/tools/libs/cr.py), maybe later HG should migrate to it, so it can be used by DI directly.